modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorFieldMapper.java

```diff
@@ -270,7 +270,9 @@ public Mapper parse(ParseContext context) throws IOException {
         }
 
         XContentParser parser = context.parser();
-        QueryBuilder queryBuilder = parseQueryBuilder(queryShardContext.newParseContext(parser), parser.getTokenLocation());
+        QueryBuilder queryBuilder = parseQueryBuilder(
+                queryShardContext.newParseContext(parser), parser.getTokenLocation()
```

Comment:
> we still need to somehow parse the incoming queries and rewrite their source before we store them no?

Luckily we already rewrite the source already at line 281

> If they come from an API we need to parse with painless as default and then rewrite the source to be explicit?

Because of the rewrite at line 281 and the fact the `lang` param is explicit set to `painless` if no `lang` has been specified in the Script class, all percolator queries from 5.0 and up will have explicit lang in any `script` query or `script` score function.

Percolator queries stored in indices created prior the ES 5 will need to be reindexed at some point after the upgrade. These percolator queries don't use the `percolator` field type and use the `object` field type, so we can't rewrite those queries when updated. However when these queries are used by the `percolate` query we pick the legacy script language (PercolatorQueryBuilder line 518). 

I think this approach is ok as we already strongly advice to reindex any 2.x percolator index in the reference documentation and migrate docs?